susePublicCloudInfoClient/lib/susepubliccloudinfoclient/infoserverrequests.py

```diff
@@ -19,30 +19,112 @@
 # <http://www.gnu.org/licenses/>.
 #
 
+import json
+import re
 import requests
+import sys
+import urllib
+import xml.etree.ElementTree as ET
+
+
+
+def __apply_filters(superset, filters):
+    # map operators to filter functions
+    filter_operations = {
+        '=': __filter_exact,
+        '~': __filter_substring,
+        '>': __filter_greater_than,
+        '<': __filter_less_than
+    }
+    # prepopulate the result set with all the items
+    result_set = superset
+    # run through the filters, allowing each to reduce the result set...
+    for a_filter in filters:
+        result_set = filter_operations[a_filter['operator']](
+            result_set,
+            a_filter['attr'],
+            a_filter['value']
+        )    
+    return result_set
+
+
+def __filter_exact(items, attr, value):
+    """select from items list where the attribute is an exact match to 'value'"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if item[attr] == value:
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_substring(items, attr, value):
+    """select from items list where 'value' is a substring of the attribute"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if value in item[attr]:
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_less_than(items, attr, value):
+    """select from items list where the attribute is less than 'value' as integers"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if int(item[attr]) < int(value):
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_greater_than(items, attr, value):
+    """select from items list where the attribute is greater than 'value' as integers"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if int(item[attr]) > int(value):
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
 
 
 def __form_url(
         framework,
         info_type,
-        data_format='xml',
-        region=None,
+        result_format='xml',
+        region='all',
```

Comment:
Now the default is being set to 'all' but if the default is used then later in the code the value is set to 'None', why not just leave the default value at 'None'. It is a keyword argument and can be skipped if so desired.

I can see that the intent, i.e. "if no region is given we get information about all regions" is better communicated when the default value is "all". However, given that this is a function of the server API rather than this code I would tend to put this into a comment rather than have the additional condition later in the code.